receiver/awsecscontainermetricsreceiver/internal/awsecscontainermetrics/metrics_helper_test.go

```diff
@@ -95,6 +95,112 @@ func TestGetContainerMetricsMissingMemory(t *testing.T) {
 	require.EqualValues(t, v, containerMetrics.StorageWriteBytes)
 }
 
+func TestGetContainerDereferenceCheck(t *testing.T) {
+
+	tests := []struct {
+		memoryStats  MemoryStats
+		testName     string
+		cpuStats     CPUStats
+		networkRate  NetworkRateStats
+		prevCpuStats CPUStats
+	}{
+		{
+			memoryStats: MemoryStats{
+				Usage:          nil,
+				MaxUsage:       nil,
+				Limit:          nil,
+				MemoryReserved: nil,
+				MemoryUtilized: nil,
+				Stats:          nil,
+			},
+			testName:     "nil memory stats values",
+			cpuStats:     cpuStats,
+			networkRate:  netRate,
+			prevCpuStats: previousCPUStats,
+		},
+		{
+			memoryStats: mem,
+			testName:    "nil cpuStats values",
+			cpuStats: CPUStats{
+				CPUUsage:       &cpuUsage,
+				OnlineCpus:     nil,
+				SystemCPUUsage: nil,
+				CPUReserved:    nil,
+				CPUUtilized:    nil,
+			},
+			networkRate:  netRate,
+			prevCpuStats: previousCPUStats,
+		},
+		{
+			memoryStats: mem,
+			testName:    "nil cpuUsage and CPUstats values",
+			cpuStats: CPUStats{
+				CPUUsage: &CPUUsage{
+					TotalUsage:        nil,
+					UsageInKernelmode: nil,
+					UsageInUserMode:   nil,
+					PerCPUUsage:       nil,
+				},
+				OnlineCpus:     nil,
+				SystemCPUUsage: nil,
+				CPUReserved:    nil,
+				CPUUtilized:    nil,
+			},
+			networkRate:  netRate,
+			prevCpuStats: previousCPUStats,
+		},
+		{
+			memoryStats: mem,
+			testName:    "nil network rate values",
+			cpuStats:    cpuStats,
+			networkRate: NetworkRateStats{
+				RxBytesPerSecond: nil,
+				TxBytesPerSecond: nil,
+			},
+			prevCpuStats: previousCPUStats,
+		},
+		{
+			memoryStats: mem,
+			testName:    "nil prev cpu stats values",
+			cpuStats:    cpuStats,
+			networkRate: netRate,
+			prevCpuStats: CPUStats{
+				CPUUsage: &CPUUsage{
```

Comment:
Can you add a test where this field is `nil`?  It's not clear that all access to this is safe.

Reply:
Good catch! You found another one! Gotta catch em all!

Reply:
Added some additional tests and nil checks!